src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -135,26 +148,77 @@ const Panel: React.FC<SelectPanelProps> = ({
     [internalOpen],
   )
 
+  /* Dialog */
+  const dialogRef = React.useRef<HTMLDialogElement>(null)
+  if (internalOpen) dialogRef.current?.showModal()
+  else dialogRef.current?.close()
+
+  // dialog handles Esc automatically, so we have to sync internal state
+  React.useEffect(() => dialogRef.current?.addEventListener('close', onInternalClose))
+
+  // React doesn't support autoFocus for dialog: https://github.com/facebook/react/issues/23301
+  // tl;dr: react takes over autofocus instead of letting the browser handle it,
+  // but not for dialogs, so we have to do it
+  React.useEffect(() => {
+    if (internalOpen) document.querySelector('input')?.focus()
+  }, [internalOpen])
+
+  /* Anchored */
+  const {position} = useAnchoredPosition(
+    {
+      anchorElementRef: anchorRef,
+      floatingElementRef: dialogRef,
+      side: 'outside-bottom',
+      align: 'start',
+    },
+    [anchorRef.current, dialogRef.current],
+  )
+
+  /* 
+    We don't close the panel when clicking outside.
+    For many years, we used to save changes and closed the dialog (for label picker)
+    which isn't accessible, clicking outside should discard changes and close the dialog
+    Fixing this a11y bug would confuse users, so as a middle ground,
+    we don't close the menu and nudge the user towards the footer actions
+  */
+  const [footerAnimationEnabled, setFooterAnimationEnabled] = React.useState(false)
+  const onClickOutside = () => {
+    setFooterAnimationEnabled(true)
+    window.setTimeout(() => setFooterAnimationEnabled(false), 500)
+  }
+
   return (
     <>
-      <AnchoredOverlay
-        anchorRef={anchorRef}
-        renderAnchor={renderAnchor}
-        open={internalOpen}
-        onOpen={() => setInternalOpen(true)}
-        onClose={onInternalClose}
+      {Anchor}
+
+      <StyledOverlay
+        as="dialog"
+        ref={dialogRef}
+        aria-labelledby={`${panelId}--title`}
+        aria-describedby={description ? `${panelId}--description` : undefined}
         width={width}
         height={height}
-        focusZoneSettings={{
-          // we only want focus trap from the overlay,
-          // we don't want focus zone on the whole overlay because
-          // we have a focus zone on the list
-          disabled: true,
+        sx={{
+          ...position,
+          // reset dialog default styles
+          border: 'none',
+          padding: 0,
+          margin: 0,
+          '::backdrop': {background: 'transparent'},
+
+          '& [data-selectpanel-primary-actions]': {
+            animation: footerAnimationEnabled ? 'selectpanel-gelatine 0.5s linear' : 'none',
```

Comment:
I think this feels a little nicer when we speed it up to `350ms`

Reply:
Done!

Here's a video to compare durations:

https://github.com/primer/react/assets/1863771/44c8ba47-c05a-4d4c-9747-4a0a64abb48c

Reply:
Love that you actually did this in the end! ❤️

Reply:
I couldn't see it in the code but is it disabled when reduced motion is on?